### ISSUE TITLE
Thought-Provoking Stories list does not indicate its size and position to VoiceOver

### ISSUE THREAD

**Issue by anna-yeddi** [mozilla-mobile/firefox-ios#25589]
## Steps to reproduce
With a VoiceOver running, navigate to any story under the "Thought-provoking stories" and observe the VO announcement 

### Expected behavior
1. VoiceOver should announce the total length of the list and the position of this specific item

### Actual behavior
1. VoiceOver does not announce any information about the list length and current item's position in it

#### Note: 
Seems that there were changes in the iOS (or VoiceOver) handling of lists - this information is no longer announced by the VoiceOver on iOS-own apps like App Store > Games tabs... Marking it as a "Feature" - it could be an enhancement to indicate for users how long the list is and where they're at. 

If we customize the announcement, the list length is usually announced at the entry of the list (i.e. when VO lands the focus on the first item of the list while navigating forward or on the last item, when navigating backwards). Default behavior was similar to the current Tab announcements (refer to the iOS App store tabs at the bottom of the view) with `1 of 4` and similar information provided for each item

### Device & build information
* Device: iPhone 15 Pro, iOS 18.3.2
* Build version: 9000 (52434)

*# 
*## Notes
Attachments: 
<
!--- Screenshots or screen recordings are very helpful for reproducing|width=200,height=183!
 -->

![width=200,height=183](https://github.com/user-attachments/assets/19f23b10-395b-4d87-a179-304115a74d11)

## :information_source:  Reference Person
@cyndichin

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-11735)

**Comment by yusuphjoluwasen** [comment]
Hi @cyndichin, I'd like to work on this issue

**Comment by lmarceau** [comment]
I think this can be announced with an `accessibilityHint` on the cell itself. A new string in our `Strings.swift` file should be added for this (with a value of `%@ of %@`). I assigned the issue to you @yusuphjoluwasen, let us know if there's anything!
